pallets/api/src/fungibles/mod.rs

```diff
@@ -159,14 +159,38 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		/// Reads the state of the fungible asset based on the provided key.
+		///
+		/// This function matches the key to determine the type of state query and returns the
+		/// encoded result.
+		///
+		/// # Arguments
+		/// * `key` - An instance of `FungiblesKey<T>`, which specifies the type of state query and
+		/// 		  the associated parameters.
+		///
+		/// # Returns
+		/// A vector of bytes representing the encoded result of the state query.
+		pub fn read_state(key: FungiblesKey<T>) -> Vec<u8> {
+			use FungiblesKey::*;
+
+			match key {
+				TotalSupply(id) => Self::total_supply(id).encode(),
```

Comment:
Probably fine for now, but querying scale encoded state, to decode it, to reencode is inefficient. A raw storage read might be better, but the downside is not benefiting from the assets pallet API protecting against inner changes. Unit tests might resolve this gap.

Reply:
I have tried this out with `total_supply` but it brings other difficulties with it. For this case I am querying `AssetDetails` from `Asset` storage item and need to obtain the value from field [supply](https://github.com/paritytech/polkadot-sdk/blob/18db502172bdf438f086cd5964c646b318b8ad37/substrate/frame/assets/src/types.rs#L62). 

The flow is creating a raw key and obtaining an encoded `AssetDetails` value. This then has to be stripped manually in order to obtain the specific bytes to decode into the `supply` value. Before I implement this I want a second opinion about whether this is worth it because it feels very error prone.

I also can't decode the `AssetDetails` value to easily obtain the supply value because `supply` is not public. In addition, this comes back to the same - decoding and encoding the value.

Reply:
Yeah, overkill for a nested field. We can revisit one day perhaps. Thanks for looking at it in more detail.